From ee15d975fdb435096224a0128dd2f48e7ff07a39 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Tue, 13 Dec 2016 14:36:16 -0600
Subject: [PATCH 045/205] soc: ti: pm33xx: Use cpu_idle_poll_ctrl in suspend
 path

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

Call cpu_idle_poll_ctrl at beginning and end of suspend path to avoid
races between cpuidle and suspend trying to communicate with the
wkup_m3, during suspend we only want it configured for entry to suspend.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
(cherry picked from commit abf3b413d03e67d23c4e0a9b8386eb8c5977e462)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/soc/ti/pm33xx.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/soc/ti/pm33xx.c b/drivers/soc/ti/pm33xx.c
index 7cff3bc..21d30f0 100644
--- a/drivers/soc/ti/pm33xx.c
+++ b/drivers/soc/ti/pm33xx.c
@@ -102,6 +102,8 @@ static int am33xx_pm_begin(suspend_state_t state)
 {
 	int ret = -EINVAL;
 
+	cpu_idle_poll_ctrl(true);
+
 	switch (state) {
 	case PM_SUSPEND_MEM:
 		ret = m3_ipc->ops->prepare_low_power(m3_ipc, WKUP_M3_DEEPSLEEP);
@@ -117,6 +119,8 @@ static int am33xx_pm_begin(suspend_state_t state)
 static void am33xx_pm_end(void)
 {
 	m3_ipc->ops->finish_low_power(m3_ipc);
+
+	cpu_idle_poll_ctrl(false);
 }
 
 static int am33xx_pm_valid(suspend_state_t state)
-- 
1.7.9.5

